src/spaces/productspace.jl

```diff
@@ -271,19 +278,25 @@ function insertleftunit(
 end
 
 """
-    insertrightunit(P::ProductSpace, i=lenght(P); conj=false, dual=false)
+    insertrightunitspace(P::ProductSpace, i=length(P); conj=false, dual=false)
 
 Insert a trivial vector space, isomorphic to the underlying field, after position `i`,
 which can be specified as an `Int` or as `Val(i)` for improved type stability.
 More specifically, adds a right monoidal unit or its dual.
 
-See also [`insertleftunit`](@ref insertleftunit(::ProductSpace, ::Val{i}) where {i}), [`removeunit`](@ref removeunit(::ProductSpace, ::Val{i}) where {i}).
+See also [`insertleftunitspace`](@ref insertleftunitspace(::ProductSpace, ::Val{i}) where {i}), [`removeunitspace`](@ref removeunitspace(::ProductSpace, ::Val{i}) where {i}).
 """
-function insertrightunit(
-        P::ProductSpace, ::Val{i} = Val(length(P));
-        conj::Bool = false, dual::Bool = false
-    ) where {i}
-    u = unitspace(spacetype(P))
+function insertrightunitspace(P::ProductSpace, ::Val{i}=Val(length(P));
+                         conj::Bool=false, dual::Bool=false) where {i}
+    N = length(P)
+    I = sectortype(P)
+    if UnitStyle(I) isa SimpleUnit
+        u = unitspace(spacetype(P))
+    else
+        N > 0 || throw(ArgumentError("cannot insert a sensible unit space in the empty product space"))
+        i == 0 && throw(DomainError((P, i), "cannot insert a sensible right unit space"))
+        u = rightunitspace(P[i])
+    end
     if dual
         u = TensorKit.dual(u)
```

Comment:
Not sure why the `TensorKit.` was here.
```suggestion
        u = dual(u)
```